atomic_reactor/plugins/post_pulp_tag.py

```diff
@@ -0,0 +1,85 @@
+    directed by tag_conf. Raise an error if two tags have that annotation.
+
+    """
+
+    key = PLUGIN_PULP_TAG_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, pulp_registry_name, pulp_secret_path=None,
+                 username=None, password=None, dockpulp_loglevel=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param pulp_registry_name: str, name of pulp registry to use, specified in /etc/
+                                   dockpulp.conf
+        :param pulp_secret_path: path to pulp.cer and pulp.key; $SOURCE_SECRET_PATH otherwise
+        :param username: pulp username, used in preference to certificate and key
+        :param password: pulp password, used in preference to certificate and key
+        """
+        # call parent constructor
+        super(PulpTagPlugin, self).__init__(tasker, workflow)
+        self.pulp_registry_name = pulp_registry_name
+        self.pulp_secret_path = pulp_secret_path
+        self.username = username
+        self.password = password
+
+        self.dockpulp_loglevel = dockpulp_loglevel
+
+    def set_v1_tag(self, v1_image_id):
+        image_names = self.workflow.tag_conf.images[:]
+        handler = PulpHandler(self.workflow, self.pulp_registry_name, self.log,
+                              pulp_secret_path=self.pulp_secret_path, username=self.username,
+                              password=self.password, dockpulp_loglevel=self.dockpulp_loglevel)
+
+        pulp_repos = handler.create_dockpulp_and_repos(image_names)
+        repo_tags = {}
+        for repo_id, pulp_repo in pulp_repos.items():
+            handler.update_repo(repo_id, {"tag": "%s:%s" % (",".join(pulp_repo.tags), v1_image_id)})
+            repo_tags[repo_id] = {"tag": "%s:%s" % (",".join(pulp_repo.tags), v1_image_id)}
+        return repo_tags
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+
+        worker_builds = self.workflow.build_result.annotations['worker-builds']
+        has_v1_image_id = None
+        repo_tags = {}
+
+        for platform in worker_builds:
+            build_info = get_worker_build_info(self.workflow, platform)
+            annotations = build_info.build.get_annotations()
+            v1_image_id = annotations.get('v1-image-id')
+            if v1_image_id:
+                if has_v1_image_id:
+                    msg = "two platforms with v1-image-ids: {0} and {0}".format(platform,
+                                                                                has_v1_image_id)
+                    raise RuntimeError(msg)
+                has_v1_image_id = platform
+                self.log.info("tagging v1-image-id %s for platform %s", v1_image_id, platform)
+                ret_val = self.set_v1_tag(v1_image_id)
+                if ret_val:
+                    repo_tags = ret_val
+        return repo_tags
```

Comment:
In the unit tests, you mean? You can tell flexmock to expect updateRepo to be called exactly once.